ipet/evaluation/IPETEvalTable.py

```diff
@@ -1733,5 +1740,36 @@ def aggregateTable(self, df):
 
         return pd.DataFrame(results)[columnorder]
 
+    def computeFilterResults(self, df : pd.DataFrame):
+        """Compute filter results once and spread them across filters that are equal.
+        """
+        activefilters = [f for g in self.getActiveFilterGroups() for f in g.getActiveFilters()]
+
+        # sort by name, which is a good proxy for equality
+        activefilters = sorted(activefilters, key = lambda x:x.getName())
+        for idx, f in enumerate(activefilters):
+            if idx > 0:
+                previous_f = activefilters[idx - 1]
+
+                #
+                # try to skip the computation of the filter column if an equal
+                # filter already has a stored result
+                #
+                stored_result = previous_f.getStoredResult(df)
+
+                if f.equals(previous_f) and stored_result is not None:
+                    logging.debug("Reusing previously stored result for filter {}".format(f.getName()))
+                    fcol = stored_result
+                else:
+                    fcol = f.applyFilter(df, self.getRowIndex())
```

Comment:
Very good point. Only for filters with index >= 1, a previous f should be considered.

Reply:
That makes sense. But as you do `applyFilter` here, shouldn't you also do this for the first one? (also otherwise the filter with index 1 does not have any result it could reuse)